### PR TITLE
feat: added full dimension key to meta [DHIS2-16379]

### PR DESCRIPTION
**Pull request by gnespolino**
This PR aims to add full dimension UID in metaData items
It will also add labels into metadata for enrollmentdate, incidentdate and executiondate, using their full dimension identifier (prefixed) as metaData item key.

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/params/CommonParams.java [hunk 1]
```diff
@@ -209,4 +212,25 @@ public List<DimensionIdentifier<DimensionParam>> getDimensionIdentifiers() {
   public CommonParamsDelegator delegate() {
     return new CommonParamsDelegator(getDimensionIdentifiers());
   }
+
+  /**
+   * gets all dimension identifiers, including parsed headers and order parameters, and removes
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/params/CommonParams.java:217]:
```suggestion
   * Gets all dimension identifiers, including parsed headers and order parameters, and removes
```

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/params/CommonParams.java [hunk 2]
```diff
@@ -209,4 +212,25 @@ public List<DimensionIdentifier<DimensionParam>> getDimensionIdentifiers() {
...
+   * duplicates (by getting the first element of each group) Shouldn't be used when the order of the
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/params/CommonParams.java:218]:
```suggestion
   * duplicates (by getting the first element of each group). Shouldn't be used when the order of the
```

**Review by d-bernat — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/processing/MetadataParamsHandler.java [hunk 3]
```diff
@@ -121,6 +149,61 @@ public void handle(Grid grid, CommonParams commonParams, User user, long rowsCou
     }
   }
 
+  private void addDimensionIdentifierToItemsIfNeeded(
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/processing/MetadataParamsHandler.java:152]: Please, let's add Javadoc for this method.

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/processing/MetadataParamsHandler.java [hunk 4]
```diff
@@ -121,6 +149,61 @@ public void handle(Grid grid, CommonParams commonParams, User user, long rowsCou
...
+    // "program.programstage.abc")
+    // from dimensionIdentifier as key
+    items.putAll(
+        items.entrySet().stream()
+            .filter(entry -> isSameDimension(dimensionIdentifier, entry))
+            .map(entry -> asEntryWithFullPrefix(dimensionIdentifier, entry))
+            .collect(Collectors.toMap(Entry::getKey, Entry::getValue)));
+  }
+
+  private MetadataItem getCustomLabel(DimensionIdentifier<DimensionParam> dimensionIdentifier) {
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/processing/MetadataParamsHandler.java:171]: Please, let's add Javadoc for this method.

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/processing/MetadataParamsHandler.java [hunk 5]
```diff
@@ -121,6 +149,61 @@ public void handle(Grid grid, CommonParams commonParams, User user, long rowsCou
...
+            .findFirst()
+            .map(
+                entry ->
+                    getIfNotBlankOrElse(
+                        entry.getValue().apply(dimensionIdentifier),
+                        entry.getKey().getHeaderColumnName()))
+            .orElse(null));
+  }
+
+  private String getIfNotBlankOrElse(String value, String headerColumnName) {
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/processing/MetadataParamsHandler.java:185]: I believe we can delete this method, and use
```
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#defaultIfBlank-T-T-
```
instead.

**Review by maikelarabori — APPROVED** [review]
Looks good, just some minor comments. Thx!

**Pull request merged**